tests/two_d_array_ut/two_d_array_ut.c

```diff
@@ -0,0 +1,642 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+#include <stdlib.h>
+#include <stddef.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "testrunnerswitcher.h"
+
+#include "umock_c/umock_c_negative_tests.h"
+
+#include "umock_c/umock_c.h"
+
+#define ENABLE_MOCKS
+#include "c_pal/gballoc_hl.h"
+#include "c_pal/gballoc_hl_redirect.h"
+#undef ENABLE_MOCKS
+
+#include "real_gballoc_hl.h"
+
+#include "c_pal/thandle.h"
+#include "c_util/two_d_array.h"
+
+/*TWO_D_ARRAY with regular types*/
+TWO_D_ARRAY_DEFINE_STRUCT_TYPE(uint32_t)
+THANDLE_TYPE_DECLARE(TWO_D_ARRAY_TYPEDEF_NAME(uint32_t));
+THANDLE_TYPE_DEFINE(TWO_D_ARRAY_TYPEDEF_NAME(uint32_t));
+
+TWO_D_ARRAY_TYPE_DECLARE(uint32_t);
+TWO_D_ARRAY_TYPE_DEFINE(uint32_t);
+
+typedef struct A_TEST_TAG
+{
+    int a;
+}A_TEST;
+
+THANDLE_TYPE_DECLARE(A_TEST);
+THANDLE_TYPE_DEFINE(A_TEST);
+
+TWO_D_ARRAY_DEFINE_STRUCT_TYPE(THANDLE(A_TEST))
+THANDLE_TYPE_DECLARE(TWO_D_ARRAY_TYPEDEF_NAME(THANDLE(A_TEST)));
+THANDLE_TYPE_DEFINE(TWO_D_ARRAY_TYPEDEF_NAME(THANDLE(A_TEST)));
+
+/*TWO_D_ARRAY works with THANDLEs too*/
+TWO_D_ARRAY_TYPE_DECLARE(THANDLE(A_TEST));
+TWO_D_ARRAY_TYPE_DEFINE(THANDLE(A_TEST));
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+
+BEGIN_TEST_SUITE(TEST_SUITE_NAME_FROM_CMAKE)
+
+TEST_SUITE_INITIALIZE(it_does_something)
+{
+    ASSERT_ARE_EQUAL(int, 0, real_gballoc_hl_init(NULL, NULL));
+
+    umock_c_init(on_umock_c_error);
+
+    REGISTER_GBALLOC_HL_GLOBAL_MOCK_HOOK();
+}
+
+TEST_SUITE_CLEANUP(TestClassCleanup)
+{
+    umock_c_deinit();
+
+    real_gballoc_hl_deinit();
+}
+
+TEST_FUNCTION_INITIALIZE(f)
+{
+    umock_c_negative_tests_init();
+    umock_c_reset_all_calls();
+}
+
+TEST_FUNCTION_CLEANUP(cleans)
+{
+    umock_c_negative_tests_deinit();
+}
+
+/*TWO_D_ARRAY_CREATE(T)*/
+
+/* Tests_SRS_TWO_D_ARRAY_07_003: [ TWO_D_ARRAY_CREATE(T) shall call THANDLE_MALLOC_FLEX with TWO_D_ARRAY_FREE(T) as dispose function, nmemb set to row_size and size set to sizeof(T*). ]*/
+/* Tests_SRS_TWO_D_ARRAY_07_004: [ TWO_D_ARRAY_CREATE(T) shall set all rows point to NULL. ]*/
+/* Tests_SRS_TWO_D_ARRAY_07_006: [ TWO_D_ARRAY_CREATE(T) shall succeed and return a non-NULL value. ]*/
+TEST_FUNCTION(TWO_D_ARRAY_CREATE_create_uint32_t_succeeds)
+{
+    //arrange
+    STRICT_EXPECTED_CALL(malloc_flex(IGNORED_ARG, 5, sizeof(uint32_t*)));
+
+    //act
+    TWO_D_ARRAY(uint32_t) tdarr = TWO_D_ARRAY_CREATE(uint32_t)(5, 5);
+
+    //assert
+    ASSERT_IS_NOT_NULL(tdarr);
+    for (unsigned int i = 0; i < 5; i++)
```

Comment:
>unsigned int [](http://example.com/codeflow?start=9&length=12)

We usually use uint32_t (the fixed size types)

Reply:
changed to use uint32_t here and everywhere else